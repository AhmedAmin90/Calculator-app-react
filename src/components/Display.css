.Display {
  background-color: #858693;
  height: 100px;
  font-weight: bold;
  padding: 0 10px;
  color: white;
  display: flex;
  flex-direction: row-reverse;
  align-items: center;
}
