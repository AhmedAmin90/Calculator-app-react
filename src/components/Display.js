import React from 'react';
import PropTypes from 'prop-types';
import './Display.css';

const Display = ({ result }) => (
  <div className="Display">
    <h1>{result}</h1>
  </div>
);

Display.defaultProps = {
  result: '0',
};

Display.propTypes = {
  result: PropTypes.string,
};

export default Display;
