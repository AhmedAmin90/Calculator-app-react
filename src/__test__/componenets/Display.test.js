/* eslint-disable no-unused-expressions */
import React from 'react';
import { create } from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Display from '../../components/Display';

test('It takes result prop and render correctly', () => {
  render(<Display result="5" />);
  const elem = create(<Display result="5" />);
  expect(elem).toMatchSnapshot();
  const displayResult = screen.getByText('5');
  expect(displayResult).toBeInTheDocument;
  expect(displayResult).toBeString;
});
