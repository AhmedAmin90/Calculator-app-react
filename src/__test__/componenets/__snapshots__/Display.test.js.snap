exports[`test It takes result prop and render correctly 1`] = `
<div
  className="Display">
  <h1>
    5
  </h1>
</div>
`;
